crawler.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import "fmt"
+
+func Crawl(url string, depth int, fetcher Fetcher) {
```

Comment:
Crawl uses fetcher to recursively crawl pages starting with url, to a maximum of depth.